### PR TITLE
Fix export default expression comment

### LINKED ISSUE #48305: JSDoc disappears after compilation
# Bug Report

### 🔎 Search Terms

JSDoc
### 🕗 Version & Regression Information

typescript version 4.7.0-dev.20220316

### 💻 Code

```ts
// file: src/symbol.ts
export const sym1 = Symbol()
export const sym2 = Symbol()
```

```ts
// file: src/export.ts

import { sym1, sym2 } from './symbol';

/** # Comments for namedExport */
export const namedExport = [sym1] as const;

/** # Comments for defaultExport */
export default [sym2] as const;
```

```ts
// file: src/import.ts
import defaultExport, { namedExport } from './export';

export { defaultExport, namedExport };
```

In ts source code, it works well.

image
image

---

If I build it by tsc:

```ts
// file: types/symbol.d.ts
export declare const sym1: unique symbol;
export declare const sym2: unique symbol;
```

```ts
// file: types/export.d.ts
import { sym1, sym2 } from './symbol';
/** # Comments for namedExport */
export declare const namedExport: readonly [typeof sym1];
declare const _default: readonly [typeof sym2];
/** # Comments for defaultExport */
export default _default;
```

```ts
// file: types/import.d.ts
import defaultExport, { namedExport } from './export';
export { defaultExport, namedExport };
```

image
image

JSDoc of defaultExport disappeared.

---

This feature is import for vue users. Because most of `.vue` file use `export default defineComponent({});`.

### PR DESCRIPTION
**Pull request by xiaoxiangmoe**
Fixes #48305

### CODE REVIEW

**Review by Kingwl — APPROVED** [review]
LGTM

File: src/compiler/transformers/declarations.ts [hunk 1]
```diff
@@ -1140,6 +1140,9 @@ namespace ts {
                         const varDecl = factory.createVariableDeclaration(newId, /*exclamationToken*/ undefined, resolver.createTypeOfExpression(input.expression, input, declarationEmitNodeBuilderFlags, symbolTracker), /*initializer*/ undefined);
                         errorFallbackNode = undefined;
                         const statement = factory.createVariableStatement(needsDeclare ? [factory.createModifier(SyntaxKind.DeclareKeyword)] : [], factory.createVariableDeclarationList([varDecl], NodeFlags.Const));
+
+                        preserveJsDoc(statement, input);
+                        removeAllComments(input);
```
Comment (sandersn) [inline src/compiler/transformers/declarations.ts:1145]: this is probably a question for @rbuckton , but why do you need to `removeAllComments(input)`? The other uses of preserveJSDoc don't do that.

**Comment by sandersn** [comment]
I have a question about why removeAllComments is needed.

**Comment by Kingwl** [comment]
> I have a question about why removeAllComments is needed.

The `Input` is still in the output code. And the `preserveJsDoc ` will not remove the original comments.
It's will be duplicated if we don't remove the comments.

**Comment by xiaoxiangmoe** [comment]
@sandersn 

If I add `removeAllComments(input);`, it will be
```ts
/**
 * JSDoc Comments
 */
declare const _default: any;
export default _default;
```

If I remove `removeAllComments(input);`, it will be

```ts
/**
 * JSDoc Comments
 */
declare const _default: any;
/**
 * JSDoc Comments
 */
export default _default;
```

**Review by sandersn — APPROVED** [review]

**Pull request merged**